dds/DCPS/DiscoveryBase.h

```diff
@@ -1323,6 +1323,35 @@ namespace OpenDDS {
           reader_local = true;
         }
 
+         // for Xtypes, check consistency
+         XTypes::TypeIdentifier writer_type_id = writer_type_info.minimal.typeid_with_size.type_id;
+         XTypes::TypeIdentifier reader_type_id = reader_type_info.minimal.typeid_with_size.type_id;
+        if (writer_type_id.kind() != XTypes::TK_NONE && reader_type_id.kind() != XTypes::TK_NONE) {
+          //look up topic name
+          TopicNameMap::iterator name_iter = topic_names_.find(writer); //TODO find reader too?
+          if (name_iter != topic_names_.end()) {
+            //find TopicDetails instance in topics_
+            typename OPENDDS_MAP(OPENDDS_STRING, TopicDetails)::iterator TDiter = topics_.find(name_iter->second);
+            if (TDiter != topics_.end()) {
+              if (!reader_local || !writer_local) { //skip for writer and reader both local
+                if (!reader_local) {
+                  TDiter->second.add_pub_sub_xtype(reader, writer_type_id, reader_type_id);
+                }
+
+                if (!writer_local) {
+                  TDiter->second.add_pub_sub_xtype(writer, reader_type_id, writer_type_id);
+                }
+              }
+            }
+
+          } else {
+            //something bad happened TODO what to do here?
```

Comment:
> Return an error and describe the error.
> 
> ACE_ERROR((LM_ERROR,
> ACE_TEXT("(%P|%t) ERROR: DiscoveryBase::match_continue ")
> ACE_TEXT("consistency check failed!\n")));
> return;

I agree we should print an error and return. I'm wondering if at this point we need to check both that the reader and the writer have the same topic name. If true then there might be some more details to print in the error. I don't think I would state this as a failure in the consistency check since it's more the conditions to run the consistency check that have failed.